src/Raven.Server/Documents/Handlers/Processors/Smuggler/AbstractSmugglerHandlerProcessorForImportGet.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Smuggler;
+using Raven.Client.Exceptions.Security;
+using Raven.Server.ServerWide;
+using Raven.Server.Smuggler.Documents.Data;
+using Sparrow.Json;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents.Handlers.Processors.Smuggler;
+
+internal abstract class AbstractSmugglerHandlerProcessorForImportGet<TRequestHandler, TOperationContext> : AbstractSmugglerHandlerProcessorForImport<TRequestHandler, TOperationContext>
+    where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+    where TOperationContext : JsonOperationContext
+{
+    protected AbstractSmugglerHandlerProcessorForImportGet([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+    {
+    }
+
+    protected abstract ImportDelegate DoImport { get; }
+
+    protected abstract long GetOperationId();
+
+    protected override async ValueTask ImportAsync(JsonOperationContext context, long? operationId)
+    {
+        if (HttpContext.Request.Query.ContainsKey("file") == false &&
+            HttpContext.Request.Query.ContainsKey("url") == false)
+        {
+            throw new ArgumentException("'file' or 'url' are mandatory when using GET /smuggler/import");
+        }
+
+        operationId ??= GetOperationId();
+        await using (var file = await GetImportStream())
+        await using (var stream = new GZipStream(new BufferedStream(file, 128 * Voron.Global.Constants.Size.Kilobyte), CompressionMode.Decompress))
+        {
+            var token = RequestHandler.CreateOperationToken();
+            await DoImport(context, stream, options: null, result: null, onProgress: null, operationId.Value, token);
+        }
+    }
+
+    private async Task<Stream> GetImportStream()
+    {
+        HttpClient httpClient = new HttpClient();
```

Comment:
This is expensive, let's move this under the condition of `if (string.IsNullOrEmpty(url) == false)` and we also need to dispose it.

Reply:
Fixed